### DISCUSSION TITLE
Mouse position from imgui window

### DISCUSSION THREAD

**Discussion by jmfp** [MonoGame/MonoGame#8389] [Inputs]
I’m building a small “engine” on top of monogame and I’m using imgui.net for a GUI. The game window is rendered to a rendertarget2D and then sent to imgui

Initializing rendertarget:
`renderTarget = new RenderTarget2D(
                GraphicsDevice,
                GraphicsDevice.PresentationParameters.BackBufferWidth,
                GraphicsDevice.PresentationParameters.BackBufferHeight,
                false,
                GraphicsDevice.PresentationParameters.BackBufferFormat,
                DepthFormat.Depth24); `

binding the rendertarget texture to imgui
(Ignore render destination, it’s essentially a vec2 at 1920, 1080):
`ImGui.Image(GuiRenderer.BindTexture(renderTarget), new System.Numerics.Vector2(renderDestination.Width, renderDestination.Height));`

Drawing the scene to the render texture:
`protected void DrawSceneToTexture(RenderTarget2D renderTarget)
    {
        // Set the render target
        GraphicsDevice.SetRenderTarget(renderTarget);
        GraphicsDevice.DepthStencilState = new DepthStencilState() { DepthBufferEnable = true };
        // Draw the scene
        GraphicsDevice.Clear(Color.CornflowerBlue);
        _spriteBatch.Begin(SpriteSortMode.BackToFront, transformMatrix: camera.Transform,  samplerState: SamplerState.PointClamp);
        //draw tilemap
        TileMap tMap = currentScene.gameObjects[2].GetComponent<TileMap>() as TileMap;
        tMap.Draw(_spriteBatch);
        _spriteBatch.Draw(testTexture, new Vector2(100, 100), Color.White);
        sprites[1].Draw(_spriteBatch, new Vector2(116, 116));
        SpriteRenderer ren = currentScene.gameObjects[1].GetComponent<SpriteRenderer>() as SpriteRenderer;
        ren.Draw(_spriteBatch);
        //currentScene.gameObjects[1].components[0].Draw(_spriteBatch);
        _spriteBatch.End();
        // Drop the render target
        GraphicsDevice.SetRenderTarget(null);
    } `

I want the mouse position to be relative to just the Imgui window while the game is in the editor, basically I want the mouse position just on the rendertarget2D but it is still getting the position from the whole window. Is there a way to do this with monogame or imgui.net?

**Comment by ThomasFOG** [comment]
There are different ways to retrieve the mouse position relative to an ImGUI window, one of them being this right when you start your window:

```System.Numerics.Vector2 mousePos = ImGui.GetIO().MousePos - ImGui.GetCursorScreenPos();```

**Reply by jmfp** [reply]
This is helpful and I will try it out when I can. I should have specified that I have enabled docking in imgui and I am rendering the render target from monogame onto an ImGui.Image. Which doesn’t take up the entire window. 

What I am trying to do is get monogame to recognize the top left of the rendertarget as mouse position 0, 0 but it is still calling the top left of the whole window 0,0 because monogame isn’t expecting imgui to take up real estate.

**Reply by ThomasFOG** [reply]
You can't do that with MonoGame's Mouse class. You'll have to roll your own mouse code.
